cirq-core/cirq/ops/conditional_operation.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import (
+    AbstractSet,
+    Any,
+    cast,
+    Dict,
+    FrozenSet,
+    Optional,
+    Sequence,
+    TYPE_CHECKING,
+    Tuple,
+    Union,
+)
+
+from cirq import protocols, value
+from cirq.ops import raw_types
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@value.value_equality
+class ConditionalOperation(raw_types.Operation):
```

Comment:
[Bikeshed] We should vote on this name at the Cirq sync - my intuition is that "conditional" may be too semantically similar to "controlled" for users to consistently remember which is which.

Might end up being a moot point if we push for merging conditional / controlled / tagged operations into the base Operation class, but we should be clear in the meantime.

Reply:
FWIW I actually kind of like the wrapper classes now. Or maybe I've just gotten used to them.

Reply:
(unresolving so we remember to discuss at Cirq sync)